### ISSUE TITLE
[python-package] Objective parameter with custom loss

### ISSUE THREAD

**Issue by sh0gg0th** [microsoft/LightGBM#5617]
The documents say that the default `objective ` is `regression`. If I'm doing classification with a custom loss, do I need to change the `objective` to binary, or is it superseded? E.g. in
```
params['objective'] = 'binary'

model = lgbm.train(params,
                   dataset,
                   fobj=my_custom_objective_function)
```

do I need the `params['objective'] = 'binary'` part? Thanks!

**Comment by jameslamb** [comment]
Thanks for using LightGBM!

In the Python package, if you're using a custom objective function for binary classification, it's not necessary to pass `"objective": "binary"` through params.

Consider this minimal example:

```python
import lightgbm as lgb
import numpy as np
from sklearn.datasets import load_breast_cancer


def logloss_obj(preds, train_data):
    y_true = train_data.get_label()
    y_pred = 1.0 / (1.0 + np.exp(-preds))
    grad = y_pred - y_true
    hess = y_pred * (1.0 - y_pred)
    return grad, hess


X, y = load_breast_cancer(return_X_y=True)
dtrain = lgb.Dataset(X, y)

bst = lgb.train(
    params={"objective": logloss_obj},
    train_set=dtrain
)

preds_raw = bst.predict(X, raw_score=True)
pred_class = (1.0 / (1.0 + np.exp(-preds_raw))) > 0.5
print(f"accuracy: {accuracy_score(y, pred_class.astype('int'))}")
# accuracy: 0.9613
```

This is because when the objective you pass into `lightgbm.train()` is callable, the parameter `objective` passed through to LightGBM's C++ code is set to the special value `"none"`.

https://github.com/microsoft/LightGBM/blob/f0cfbff63f8a228784e52e033533e5cb3fa1b97b/python-package/lightgbm/engine.py#L137-L139

-------

In the future, when you ask a question here please provide the information we've asked for, such as:

* how you are using LightGBM (looks like the Python package but please say that explicitly)
* version of LightGBM you are using
* how you installed LightGBM

**Comment by sh0gg0th** [comment]
Understood, thanks a lot @jameslamb ! Yes it's a Python package, my bad, I will be sure to include the information in the future.
